catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -534,28 +546,111 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+          Map<String, Object> body = mapper.parser().parseMap(util.safeGetBody(req));
 
-          Map<String, Object> workspace =
-              JsonFactory.create().parser().parseMap(util.safeGetBody(req));
-
-          Metacard metacard = transformer.transform(workspace);
-          metacard.setAttribute(new AttributeImpl(Metacard.ID, id));
+          Metacard updated = transformer.transform(body);
+          updated.setAttribute(new AttributeImpl(Core.ID, id));
 
-          Metacard updated = updateMetacard(id, metacard);
-          return util.getJson(transformer.transform(updated));
+          Metacard stored = updateMetacard(id, updated);
+          return util.getJson(transformer.transform(stored));
         });
 
     delete(
         "/workspaces/:id",
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+          WorkspaceMetacardImpl workspace = workspaceService.getWorkspaceMetacard(id);
+
+          String[] queryIds = workspace.getQueries().toArray(new String[0]);
+
+          if (queryIds.length > 0) {
+            catalogFramework.delete(new DeleteRequestImpl(queryIds));
```

Comment:
❓ couldn't find if this was already asked, but does this mean that there will always be a query metacard created for each workspace query? (meaning they can't be included on other workspaces?) just making sure this query wont appear on other workspaces otherwise they'll fail to find it after this workspace (and queries) are deleted.

Reply:
I tried not to do too much future planning on this PR. There are some features coming down the pipe that will require some of this to be changed. This is one of those areas. Actually this might be the only area for shareable queries. 

I could be convinced to perform a check if any other workspaces are associated with any of these query IDs. I'm also a bit concerned about the performance of this operation.